packages/core/test/specs/data_sources/model/conditional_variables/ConditionalTraits.ts

```diff
@@ -0,0 +1,174 @@
+import { DataSourceManager, Editor } from '../../../../../src';
+import { DataVariableType } from '../../../../../src/data_sources/model/DataVariable';
+import { MissingConditionError } from '../../../../../src/data_sources/model/conditional_variables/DataCondition';
+import { ConditionalVariableType } from '../../../../../src/data_sources/model/conditional_variables/DataCondition';
+import { GenericOperation } from '../../../../../src/data_sources/model/conditional_variables/operators/GenericOperator';
+import { NumberOperation } from '../../../../../src/data_sources/model/conditional_variables/operators/NumberOperator';
+import { DataSourceProps } from '../../../../../src/data_sources/types';
+import ComponentWrapper from '../../../../../src/dom_components/model/ComponentWrapper';
+import EditorModel from '../../../../../src/editor/model/Editor';
+import { setupTestEditor } from '../../../../common';
+
+describe('TraitConditionalVariable', () => {
+  let editor: Editor;
+  let em: EditorModel;
+  let dsm: DataSourceManager;
+  let cmpRoot: ComponentWrapper;
+
+  beforeEach(() => {
+    ({ editor, em, dsm, cmpRoot } = setupTestEditor());
+  });
+
+  afterEach(() => {
+    em.destroy();
+  });
+
+  it('should add a trait with a condition evaluating to a string', () => {
+    const component = cmpRoot.append({
+      tagName: 'h1',
+      type: 'text',
+      traits: [
+        {
+          type: 'text',
+          name: 'title',
+          value: {
+            type: ConditionalVariableType,
+            condition: {
+              left: 0,
+              operator: NumberOperation.greaterThan,
+              right: -1,
+            },
+            ifTrue: 'Some title',
+          },
+        },
+      ],
+    })[0];
+
+    expect(component).toBeDefined();
+    expect(component.getTrait('title').get('value')).toBe('Some title');
```

Comment:
let's also check the component attribute model (`cmp.getAttributes().title`) and its rendered view.

Same for the test cases below